devel-tools/rpm2sysext

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash -e
+
+if [ "$#" = 0 ] || [ "$1" = "--help" ]; then
+	echo "Usage: $0 PACKAGE..."
+	echo "Convert RPMs to systemd-sysext compatible DDIs"
+	exit 0
+fi
+
+dest=
+cleanup()
+{
+	[ -n "$dest" ] || return
+	rm -rf "$dest"
+}
+trap cleanup EXIT
+
+declare -A uuid
+init_uuids()
+{
+	while read -r n u; do
+		[ "$n" = "NAME" ] && continue
+		[ "${n#usr}" = "$n" ] && continue
+		uuid[$n]=$u
+	done < <(systemd-id128 show -u)
+}
+
+uuid_for()
+{
+	local arch="${1:?}"
+	[ -n "$uuid" ] || init_uuids
+	case "$arch" in
+		x86_64) arch=x86-64 ;;
+		i?86) arch=x86 ;;
+		noarch) arch= ;;
+	esac
+	u="${uuid[usr-$arch]}"
+	echo "${u:-${uuid[usr]}}"
+}
+
+. /usr/lib/os-release
+
+for pkg in "$@"; do
+	dest=$(mktemp --tmpdir --directory rpm2sysext.XXXXXXXXXX)
+	read -r NVRA ARCH < <(rpm -qp --qf '%{NAME}-%{VERSION}-%{RELEASE}.%{ARCH} %{ARCH}\n' "$pkg")
+	rpm2cpio "$pkg" | cpio -idD "$dest"
+
+	mkdir -p "$dest"/usr/lib/extension-release.d
+	cat > "$dest/usr/lib/extension-release.d/extension-release.$NVRA" <<-EOF
+	ID=_any
+	SYSEXT_SCOPE="system initrd portable"
+	EOF
+	sqfs="$dest/sqfs"
+	mksquashfs "$dest/usr" "$sqfs" -force-uid 0 -force-gid 0 -keep-as-directory -quiet
+	read -r size blocksize < <(stat -c "%s %B" "$sqfs")
+	if [ "$blocksize" != 512 ]; then
+		echo "Block size $blocksize not supported" >&2
+		exit 1
+	fi
+	blocks=$((size/512+size%512))
```

Comment:
This looks wrong, it adds one block for each tail byte past the block boundary.

The worst case is a 511 B squashfs, which would end up as a 511 block image (256KiB)